### PR TITLE
Reader: Update dismissible buttons in in-stream recs

### PR DESCRIPTION
**Pull request by jancavan**
This PR updates the dismiss buttons in in-stream recs to match the mockups:

**Wide viewports:**
screenshot 2016-12-08 16 21 02

**Narrow viewports:**
screenshot 2016-12-08 16 20 49

Related: https://github.com/Automattic/wp-calypso/pull/9876

### CODE REVIEW

**Comment by jancavan** [comment]
In https://github.com/Automattic/wp-calypso/pull/9912/commits/8121441fe91962b89ba37babc093af9e24d59f00:

**Before:**
screenshot 2016-12-09 11 23 48

**After:**
screenshot 2016-12-09 11 29 53

**Comment by jancavan** [comment]
Also updated some overlapping that was between longer excerpts and the action buttons which might've been a regression from https://github.com/Automattic/wp-calypso/pull/9885:

**Before:**
screenshot 2016-12-09 11 19 34

**After:**
screenshot 2016-12-09 11 19 44

File: client/reader/stream/_style.scss [hunk 1]
```diff
@@ -431,41 +431,89 @@ $reader-related-card-v2-breakpoint-small: "( max-width: 535px )";
...
+			flex-grow: 1;
+			align-self: baseline;
+			justify-content: flex-start;
+		}
+
+		@include breakpoint( "<660px" ) {
+			justify-content: flex-end;
+		}
+
+		@media #{$reader-related-card-v2-breakpoint-small} {
```
Comment (samouri) [inline client/reader/stream/_style.scss]: since small and medium mostly share the same properties can we combine via:
`@media #{$reader-related-card-v2-breakpoint-medium}, @media #{$reader-related-card-v2-breakpoint-small} `

Comment (jancavan) [inline client/reader/stream/_style.scss]: @samouri I wish :) Already tried that in the past, but it throws a formatting error. Could probably create a mixin for it, but that'll have to be in a separate PR since we use these custom breakpoints in a few different places.

**Comment by samouri** [comment]
Behavior looked really good to me!
🚢

**Comment by fraying** [comment]
LGTM!
